src/operator/effectivepolicy/Syncer.go

```diff
@@ -0,0 +1,142 @@
+package effectivepolicy
+
+import (
+	"context"
+	goerrors "errors"
+	"github.com/amit7itz/goset"
+	"github.com/otterize/intents-operator/src/operator/api/v1alpha3"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/otterize/intents-operator/src/shared/serviceidresolver/serviceidentity"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type EffectivePoliciesApplier interface {
```

Comment:
This should be "effective policy applier" rather than "effective policies applier"

I forget what this concept is called in English grammar but here's an example to illustrate what I mean:
In English, do you say "train station" or "trains station", for a station where all trains are?
Similarly, this is an applier that handles effective policies, which makes it an effective policy applier.

An applier that handles trains would be a train applier, not a trains applier.